### PR TITLE
General cleanup

### PR DESCRIPTION
**Pull request by CountBleck**
Changes proposed in this pull request:
⯈ Skip fixture comparisons in tests if compiling them failed
⯈ Update esbuild
⯈ Clean up the DTS generator
⯈ Update glob
⯈ Marginally clean up the web build thingy
⯈ Add a coverage job to CI; maybe Codecov can be used on it in the future?

- [x] I've read the contributing guidelines
- [x] I've added my name and email to the NOTICE file

### CODE REVIEW

**Comment by CountBleck** [comment]
I gave up on trying to make the DTS generator faster. :(

File: scripts/build-web.js [hunk 1]
```diff
@@ -1,42 +1,43 @@
-import path from "path";
+import {dirname, join} from "path";
```
Comment (HerrCai0907) [inline scripts/build-web.js:1]:
```suggestion
import { dirname, join } from "path";
```

**Review by HerrCai0907 — APPROVED** [review]

File: scripts/build-web.js (hunk 1 above)
Comment (CountBleck) [reply scripts/build-web.js:1]: Whoops!

**Comment by HerrCai0907** [comment]
Can this PR be merged now, or you still want to update it? @CountBleck

**Comment by CountBleck** [comment]
I'm not entirely sure.

**Comment by CountBleck** [comment]
@dcodeIO do you have any interest in using Codecov?

**Review by HerrCai0907 — APPROVED** [review]
LGTM

**Comment by ia0** [comment]
I'm not sure why (not familiar with assemblyscript) but I suspect this PR broke [my workflow](https://github.com/google/wasefire/actions/runs/6810173264/job/18517963477?pr=285):

```
cd "examples/assemblyscript" && "/home/runner/work/wasefire/wasefire/scripts/wrapper.sh" "npm" "install" "--no-save" "assemblyscript"

added 3 packages in 1s

1 package is looking for funding
  run `npm fund` for details
cd "examples/assemblyscript" && "./node_modules/.bin/asc" "-o" "../../target/wasefire/applet.wasm" "-O3" "--lowMemoryLimit" "--stackSize" "16384" "--use" "abort=hello/main/abort" "--debug" "hello/main.ts"
node:internal/errors:496
    ErrorCaptureStackTrace(err);
    ^

Error [ERR_MODULE_NOT_FOUND]: Cannot find module '/home/runner/work/wasefire/wasefire/examples/assemblyscript/node_modules/assemblyscript/dist/asc.js' imported from /home/runner/work/wasefire/wasefire/examples/assemblyscript/node_modules/assemblyscript/bin/asc.js
    at __node_internal_captureLargerStackTrace (node:internal/errors:496:5)
    at new NodeError (node:internal/errors:405:5)
    at finalizeResolution (node:internal/modules/esm/resolve:327:11)
    at moduleResolve (node:internal/modules/esm/resolve:946:10)
    at defaultResolve (node:internal/modules/esm/resolve:1132:11)
    at nextResolve (node:internal/modules/esm/loader:163:28)
    at ESMLoader.resolve (node:internal/modules/esm/loader:835:30)
    at ESMLoader.getModuleJob (node:internal/modules/esm/loader:424:18)
    at ESMLoader.import (node:internal/modules/esm/loader:524:22)
    at importModuleDynamically (node:internal/modules/esm/translators:110:35)
    at importModuleDynamicallyCallback (node:internal/process/esm_loader:36:14)
    at file:///home/runner/work/wasefire/wasefire/examples/assemblyscript/node_modules/assemblyscript/bin/asc.js:29:21
    at ModuleJob.run (node:internal/modules/esm/module_job:194:25) {
  code: 'ERR_MODULE_NOT_FOUND'
}

Node.js v18.18.2
```

Notice how it's complaining about `dist/asc.js` which indeed doesn't exist, only `dist/asc.d.ts` does.

You can reproduce this with the following steps:

```shell
npm install --no-save assemblyscript
./node_modules/.bin/asc foo.ts
```

**Pull request merged**